### PR TITLE
release(2021-08-30): bump package versions

### PR DESCRIPTION
**Pull request by timtay-microsoft**
This is a special preview release for our upcoming 2.0.0 clients. There are substantial changes made to the SDK in this upcoming version, and the brief summary of them can be seen below:

## Java Deps (com.microsoft.azure.sdk.iot:iot-deps:2.0.0-preview-001)
- Trusted certificates are read from the physical device's trusted root certification authorities certificate store rather than from source.
- Removed all deprecated APIs 
- Removed unthrown exceptions from declared exceptions list for all APIs
- Removed support for parsing certificates

## Java IotHub Device Client (com.microsoft.azure.sdk.iot:iot-device-client:2.0.0-preview-001)
- Trusted certificates are read from the physical device's trusted root certification authorities certificate store rather than from source.
- Removed all deprecated APIs 
- Removed support for parsing certificates
- Renamed several APIs for consistency between Module and Device clients
- Removed unthrown exceptions from declared exceptions list for all APIs

## Java IotHub Service Client (com.microsoft.azure.sdk.iot:iot-service-client:2.0.0-preview-001)
- Trusted certificates are read from the physical device's trusted root certification authorities certificate store rather than from source.
- Removed all deprecated APIs 
- Removed unthrown exceptions from declared exceptions list for all APIs

## Java Provisioning Device Client (com.microsoft.azure.sdk.iot.provisioning:provisioning-device-client:2.0.0-preview-001)
- Trusted certificates are read from the physical device's trusted root certification authorities certificate store rather than from source.
- Removed all deprecated APIs 
- Removed unthrown exceptions from declared exceptions list for all APIs

## Java Provisioning Service Client (com.microsoft.azure.sdk.iot.provisioning:provisioning-service-client:2.0.0-preview-001)
- Trusted certificates are read from the physical device's trusted root certification authorities certificate store rather than from source.
- Removed all deprecated APIs 
- Removed unthrown exceptions from declared exceptions list for all APIs

## Java X509 Provider (com.microsoft.azure.sdk.iot.provisioning.security:x509-provider:2.0.0-preview-001)
- Removed support for parsing certificates
- Removed unthrown exceptions from declared exceptions list for all APIs

## Java TPM Provider (com.microsoft.azure.sdk.iot.provisioning.security:tpm-provider:2.0.0-preview-001)
- Removed unthrown exceptions from declared exceptions list for all APIs

## Java Security Provider (com.microsoft.azure.sdk.iot.provisioning.security:security-provider:2.0.0-preview-001)
- Trusted certificates are read from the physical device's trusted root certification authorities certificate store rather than from source.
- Removed unthrown exceptions from declared exceptions list for all APIs

**For a more exhaustive list of changes made in the jump from 1.0.0 to 2.0.0, and for a helpful FAQ section to answer common questions you may have, see [this document](https://github.com/Azure/azure-iot-sdk-java/blob/preview/major%20version%20upgrade%20migration%20guide.md).**

https://search.maven.org/artifact/com.microsoft.azure.sdk.iot/iot-device-client/2.0.0-preview-001/jar
https://search.maven.org/artifact/com.microsoft.azure.sdk.iot/iot-service-client/2.0.0-preview-001/jar
https://search.maven.org/artifact/com.microsoft.azure.sdk.iot/iot-deps/2.0.0-preview-001/jar
https://search.maven.org/artifact/com.microsoft.azure.sdk.iot.provisioning/provisioning-device-client/2.0.0-preview-001/jar
https://search.maven.org/artifact/com.microsoft.azure.sdk.iot.provisioning/provisioning-service-client/2.0.0-preview-001/jar
https://search.maven.org/artifact/com.microsoft.azure.sdk.iot.provisioning.security/x509-provider/2.0.0-preview-001/jar
https://search.maven.org/artifact/com.microsoft.azure.sdk.iot.provisioning.security/security-provider/2.0.0-preview-001/jar

Note that these preview jars have been tested against the same standard tests as our normal jars other than for Edgehub scenarios. Users may encounter bugs in this preview version if used for writing custom Edgehub modules.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -33,17 +33,17 @@
...
-        <security-provider-version>1.5.0</security-provider-version>
-        <tpm-provider-emulator-version>1.1.2</tpm-provider-emulator-version>
+        <iot-device-client-version>2.0.0-preview-001</iot-device-client-version>
+        <iot-service-client-version>2.0.0-preview-001</iot-service-client-version>
+        <iot-deps-version>2.0.0-preview-001</iot-deps-version>
+        <provisioning-device-client-version>2.0.0-preview-001</provisioning-device-client-version>
+        <provisioning-service-client-version>2.0.0-preview-001</provisioning-service-client-version>
+        <security-provider-version>2.0.0-preview-001</security-provider-version>
+        <tpm-provider-emulator-version>undefined</tpm-provider-emulator-version>
         <tpm-provider-version>1.1.3</tpm-provider-version>
```
Comment (abhipsaMisra) [inline pom.xml:43]: this lonely fellow got no updates

File: pom.xml [hunk 2]
```diff
@@ -33,17 +33,17 @@
...
-        <provisioning-service-client-version>1.9.0</provisioning-service-client-version>
-        <security-provider-version>1.5.0</security-provider-version>
-        <tpm-provider-emulator-version>1.1.2</tpm-provider-emulator-version>
+        <iot-device-client-version>2.0.0-preview-001</iot-device-client-version>
+        <iot-service-client-version>2.0.0-preview-001</iot-service-client-version>
+        <iot-deps-version>2.0.0-preview-001</iot-deps-version>
+        <provisioning-device-client-version>2.0.0-preview-001</provisioning-device-client-version>
+        <provisioning-service-client-version>2.0.0-preview-001</provisioning-service-client-version>
+        <security-provider-version>2.0.0-preview-001</security-provider-version>
+        <tpm-provider-emulator-version>undefined</tpm-provider-emulator-version>
```
Comment (abhipsaMisra) [inline pom.xml:42]: undefined?

Comment (timtay-microsoft) [reply pom.xml:42]: Whoops, I deleted this version from our versions.json file since this isn't a package we publish. I'll fix this

**Review by abhipsaMisra — COMMENTED** [review]
Three of the jars (tpm and two of dice) have versions specified as "undefined". What does that mean?

File: pom.xml (hunk 1 above)
Comment (timtay-microsoft) [reply pom.xml:43]: That is odd. I'll look into why this wasn't bumped, too

**Comment by timtay-microsoft** [comment]
Looks like the bump version script didn't take kindly to me removing a few unused packages and their versions from our config file. I've fixed the issue locally for this PR, but I'm looking into fixing the bump versions script as well in parallel

**Review by abhipsaMisra — APPROVED** [review]
So the two emulator packages and the dice provider are no longer required, correct?

**Comment by timtay-microsoft** [comment]
> So the two emulator packages and the dice provider are no longer required, correct?

Correct. The Dice provider was the predecessor to the x509 provider, so we opted to never ship it. The two emulators are not something that users should use in production applications, so we don't ship them either

**Pull request merged**